chrome/content/zotero/itemTreeColumns.jsx

```diff
@@ -23,36 +23,37 @@
 	***** END LICENSE BLOCK *****
 */
 
-(function() {
 const React = require('react');
 const Icons = require('components/icons');
 
 /**
- * @type Column {
- * 	dataKey: string,				// Required, see use in ItemTree#_getRowData()
- *
- * 	defaultIn: Set<string>,			// Types of trees the column is default in. Can be [default, feed];
- * 	disabledIn: Set<string>,		// Types of trees where the column is not available
- * 	defaultSort: number				// Default: 1. -1 for descending sort
- *
- * 	flex: number,					// Default: 1. When the column is added to the tree how much space it should occupy as a flex ratio
- * 	width: string,					// A column width instead of flex ratio. See above.
- * 	fixedWidth: boolean				// Default: false. Set to true to disable column resizing
- * 	staticWidth: boolean			// Default: false. Set to true to prevent columns from changing width when
- * 									// the width of the tree increases or decreases
- * 	minWidth: number,				// Override the default [20px] column min-width for resizing
- *
- * 	label: string,					// The column label. Either a string or the id to an i18n string.
- * 	iconLabel: React.Component,		// Set an Icon label instead of a text-based one
- *
- * 	ignoreInColumnPicker: boolean	// Default: false. Set to true to not display in column picker.
- * 	submenu: boolean,				// Default: false. Set to true to display the column in "More Columns" submenu of column picker.
- *
- * 	primary: boolean,				// Should only be one column at the time. Title is the primary column
- * 	zoteroPersist: Set<string>,		// Which column properties should be persisted between zotero close
- * 	}
+ * @typedef ItemTreeColumnOption
+ * @type {object}
+ * @property {string} dataKey - Required, see use in ItemTree#_getRowData()
+ * @property {Set.<string>} [defaultIn] - Types of trees the column is default in. Can be [default, feed];
```

Comment:
@adomasven 

> This API however puts things on their head. Now there are two sources of truth for what columns the ItemTree should display. The ones passed into it upon instantiation, and the ones added via the plugin API. We will need to reconcile that somehow.

Regarding this, is these two properties of the column option used for controlling that? To make it simple, we just give ALL columns upon instantiation and let the item tree itself decide whether to display a column or not depending on who it is (feed, search, etc.)

Reply:
It could be an option to make the ItemTree not have a prop for columns and just get it from some centralized location, where both default and plugin columns are stored. However both plugin developers and ourselves might want to create (and I am creating) custom ItemTree views, with custom columns, and we have to support that. See https://snipboard.io/Ami4sE.jpg for an example. So in one way or another, upon instantiation of the ItemTree you need to be able to select both default columns and add any custom columns, that are not necessary plugin-based, or persistent in any way.

Reply:
Done

Reply:
Done.

Use `extraColumns` and `columnsFilter`, instead of constantly setting the initial value of `columns` in the props of ItemTree initialization allows temporary columns and enables/disables columns dynamically.